lister.cc

```diff
@@ -75,3 +77,44 @@ future<> lister::rmdir(fs::path dir) {
         return remove_file(dir.native());
     });
 }
+
+future<std::optional<directory_entry>> queue_lister::get() {
+    if (!_done) {
+        auto f = co_await open_checked_directory(general_disk_error_handler, _dir.native());
+        auto walker = [this] (fs::path dir, directory_entry de) {
+            return _q.push_eventually(std::make_optional<directory_entry>(std::move(de))).handle_exception([] (std::exception_ptr) {
+                // ignore exception, queue was aborted
+            });
+        };
+        assert(!_lister);
```

Comment:
Nitpick: the `if(!_done)` above was weird to read (it sounds like "not done", but it's not what being tested). And here you have an `assert(!_lister)`. Wouldn't the code be easier to understand if instead you had instead of `if (!_done)` a  `if (!_lister)`?

Reply:
It was `if (!_lister)` in an earlier message, but I want `close()` to get rid of `_lister` and then if `get()` is called after `close()` I don't want it to re-create the `_lister`.  I could add a `bool _closed` member to prevent, but instead I overloaded the `_done` future that starts disengaged and is engaged once when the `_lister` is first created, and possibly on `close()` it is set to a ready future, but remains engaged.  Maybe it could be named better so it won't look weird.

Reply:
I see. I actually understood that at the end of my review but wrote this comment in the middle.
So ok. The code seems correct, I guess there are seven other ways to write it, some will look more natural to me, but nevermind.